plutus-ledger-api/src/PlutusLedgerApi/Internal/SerialisedScript.hs

```diff
@@ -0,0 +1,115 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE DeriveAnyClass  #-}
+{-# LANGUAGE TemplateHaskell #-}
+module PlutusLedgerApi.Internal.SerialisedScript
+    ( SerialisedScript
+    , scriptCBORDecoder
+    , ScriptForExecution (..)
+    , ScriptDecodeError (..)
+    , AsScriptDecodeError (..)
+    , deserialiseScriptForExecution
+    ) where
+
+import PlutusCore
+import PlutusLedgerApi.Common.Versions
+import PlutusLedgerApi.Internal.VersioningRules
+import UntypedPlutusCore qualified as UPLC
+
+import Codec.CBOR.Decoding qualified as CBOR
+import Codec.CBOR.Extras
+import Codec.CBOR.Read qualified as CBOR
+import Control.Arrow ((>>>))
+import Control.Exception
+import Control.Lens
+import Control.Monad.Error.Lens
+import Control.Monad.Except
+import Data.ByteString.Lazy as BSL (ByteString, fromStrict)
+import Data.ByteString.Short
+import Data.Coerce
+import Data.Set as Set
+import Prettyprinter
+
+data ScriptDecodeError =
+      CBORDeserialiseError CBOR.DeserialiseFailure
+    | RemainderError BSL.ByteString
+    | LanguageNotAvailableError
+        { sdeAffectedLang :: PlutusVersion
+        , sdeIntroPv      :: ProtocolVersion
+        , sdeCurrentPv    :: ProtocolVersion
+        }
+    deriving stock (Eq, Show)
+    deriving anyclass Exception
+makeClassyPrisms ''ScriptDecodeError
+
+-- | Scripts to the ledger are serialised bytestrings.
+type SerialisedScript = ShortByteString
+
+-- | A variant of `Script` with a specialized decoder.
```

Comment:
Sorry, we no longer have `Script`, and I should have updated this comment when I removed it. You may say something like "this wraps a Program with NamedDeBruijn, so it is ready for execution".